### ISSUE TITLE
Failed test on 'Test_hlset'

### ISSUE THREAD

**Issue by vp1981** [vim/vim#9100]
### Steps to reproduce

1. Clone repository
2. Run 'make'
3. Run 'make check'

(Actually I'm using `PKGBUILD` to build vim package for Archlinux and building fails on test introduced in commin d1a8d658e1b16cf8579fc72cf7aa6a29a57ff5ef)

### Expected behaviour

All tests are passed.

### Operating system

Archlinux x86_64

### Version of Vim

git

### Logs and stack traces

```shell
xecuted:  4737 Tests
 Skipped:   104 Tests
  FAILED:     1 Tests


Failures:
        From test_highlight.vim:
        Found errors in Test_hlset():
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[455]..function RunTheTest[44]..Test_hlset[104]..CheckLegacyAndVim9Success[1]..CheckTransLegacySuccess[8]..CheckLegacySuccess[7]..Func line 8: Expected ['myhlg2         xxx term=underline,strikethrough cterm=underline,strikethrough', '                   gui=underline,strikethrough'] but got ['myhlg2         xxx term=underline,strikethrough cterm=underline,strikethrough gui=underline,strikethrough']
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[455]..function RunTheTest[44]..Test_hlset[104]..CheckLegacyAndVim9Success[2]..CheckTransDefSuccess[7]..CheckDefSuccess[7]..Func line 8: Expected ['myhlg2         xxx term=underline,strikethrough cterm=underline,strikethrough', '                   gui=underline,strikethrough'] but got ['myhlg2         xxx term=underline,strikethrough cterm=underline,strikethrough gui=underline,strikethrough']
        command line..script /home/vladimir/pkgs/vim/src/vim/src/testdir/runtest.vim[455]..function RunTheTest[44]..Test_hlset[104]..CheckLegacyAndVim9Success[3]..CheckTransVim9Success[7]..CheckScriptSuccess[6]..script /home/vladimir/pkgs/vim/src/vim/src/testdir/XScriptSuccess18 line 9: Expected ['myhlg2         xxx term=underline,strikethrough cterm=underline,strikethrough', '                   gui=underline,strikethrough'] but got ['myhlg2         xxx term=underline,strikethrough cterm=underline,strikethrough gui=underline,strikethrough']

TEST FAILURE
make[2]: *** [Makefile:49: report] Error 1
make[2]: Leaving directory '/home/vladimir/pkgs/vim/src/vim/src/testdir
```

**Comment by vp1981** [comment]
Seems something fishy with spaces.

**Comment by dpelle** [comment]
I see the same bug on xubuntu-18.04.6 (clang-11), 8.2.3582, huge, nogui.

I found that the test:

* always succeeds if the terminal has <= 105 columns
* always fails if the terminal has > 105 columns

I suppose that the test needs to do `:set columns=…` to be reliable.  I'll confirm and create a PR if that fixes it.

**Comment by dpelle** [comment]
To @vp1981: I created PR https://github.com/vim/vim/pull/9101 to fix this.
Can you confirm that it also fixes it for you?

**Comment by vp1981** [comment]
Hello dpelle.
Yes, I have `COLUMNS=140` and with pr#9101 vim from git compiles fine.

**Comment by vim-ml** [comment]
* Dominique Pellé ***@***.***> [2021-11-06 01:42 -0700]:

> To @vp1981: I created PR https://github.com/vim/vim/pull/9101 to fix this.
> Can you confirm that it also fixes it for you?

Your Patch fixes the issue for me too -> $COLUMNS=238.

Thanks for providing the patch ;-)

Elimar
-- 
  On the keyboard of life you have always
  to keep a finger at the escape key;-)

**Comment by vim-ml** [comment]
Hi,

On Sat, Nov 6, 2021 at 1:42 AM Dominique Pellé ***@***.***>
wrote:

> To @vp1981 <https://github.com/vp1981>: I created PR #9101
> <https://github.com/vim/vim/pull/9101> to fix this.
> Can you confirm that it also fixes it for you?
>
>
>
I have updated this test in PR #9117 to remove the dependency on the
terminal width.

- Yegappan

**Comment by hemenge133** [comment]
Confirming that #9101 fixed the issue for me on Ubuntu 20.04.3 LTS
